src/crypto/public_key/elliptic/curves.js

```diff
@@ -27,107 +27,210 @@
 
 'use strict';
 
-import {ec as EC} from 'elliptic';
-import {KeyPair} from './key.js';
-import BigInteger from '../jsbn.js';
-import enums from '../../../enums.js';
-import util from '../../../util.js';
+import { ec as EC, eddsa as EdDSA } from 'elliptic';
+import { KeyPair } from './key';
+import BigInteger from '../jsbn';
+import random from '../../random';
+import config from '../../../config';
+import enums from '../../../enums';
+import util from '../../../util';
+import base64 from '../../../encoding/base64';
+
+const webCrypto = util.getWebCrypto();
+const nodeCrypto = util.getNodeCrypto();
+
+var webCurves = {}, nodeCurves = {};
+if (webCrypto && config.use_native) {
+  webCurves = {
+    'p256': 'P-256',
+    'p384': 'P-384',
+    'p521': 'P-521'
+  };
+} else if (nodeCrypto && config.use_native) {
+  var knownCurves = nodeCrypto.getCurves();
+  nodeCurves = {
+    'secp256k1': knownCurves.includes('secp256k1') ? 'secp256k1' : undefined,
+    'p256': knownCurves.includes('prime256v1') ? 'prime256v1' : undefined,
+    'p384': knownCurves.includes('secp384r1') ? 'secp384r1' : undefined,
+    'p521': knownCurves.includes('secp521r1') ? 'secp521r1' : undefined
+    // TODO add more here
+  };
+}
 
 const curves = {
   p256: {
     oid: util.bin2str([0x2A, 0x86, 0x48, 0xCE, 0x3D, 0x03, 0x01, 0x07]),
-    curveName: 'P-256',
-    hashName: 'SHA-256',
+    keyType: enums.publicKey.ecdsa,
     hash: enums.hash.sha256,
     cipher: enums.symmetric.aes128,
-    nist: true
+    node: nodeCurves.secp256r1,
+    web: webCurves.secp256r1,
```

Comment:
What is going on with these node and web values (for all of the NIST curves)? None of them seem to correspond to the names that node/web crypto use for the curves or the values in webCurves/nodeCurves.

Reply:
Ahh, this is definitely a bug. I'll fix it in the next commit. Thanks!